src/helpers/airswap.ts

```diff
@@ -1,18 +1,18 @@
-import { scrapeToken as airSwapScrapeToken } from '@airswap/metadata';
+import { getTokenFromContract as airswapGetTokenFromContract } from '@airswap/metadata';
 import { TokenInfo } from '@airswap/typescript';
 import * as ethers from 'ethers';
 
-export const scrapeToken = (
+export const getTokenFromContract = (
   address: string,
-  provider: ethers.providers.BaseProvider | string | null,
-  chainId?: number,
+  provider: ethers.providers.BaseProvider,
+  id?: string,
   // eslint-disable-next-line no-async-promise-executor
```

Comment:
There's a reason for this rule, IIRC it's to do with the promise never being rejected if there's an error?.. If this doesn't matter here then add a comment to say why.

BUT, why does this function exist? Seems like it doesn't add much value.

Reply:
I realise now you don't reject when undefined.. It's an odd rule, odd use of a promise, explain your actions in a comment

Reply:
You're right. This was a helper I copied from airswap-web. Will remove it.